src/components/Hangar/TankStatsComponent.js

```diff
@@ -7,25 +7,23 @@ import OverviewCard from '../StatisticsCard/Overview';
 import Damage from '../StatisticsCard/Damage';
 import Wins from '../StatisticsCard/Wins';
 import Battles from '../StatisticsCard/Battles';
-import { Filter } from './Filter';
-import { expressURL } from '../../shared/expressURL';
 import BattleStyle from '../StatisticsCard/BattleStyle';
+import { Filter } from './Filter';
 
 const TankStats = (props) => {
   const [playerStats, setPlayerStats] = useState(props.tanksStats);
+  const [filteredTankStats, setFilteredTankStats] = useState(undefined);
   const { accountId, wotId } = useParams();
   const navigate = useNavigate();
 
   useEffect(() => {
     if (playerStats?.data) return;
     (async () => {
-      const stats = await fetch(`${expressURL}tanks/${accountId}`).then((res) => res.json());
-      props.setTanksStatsData(stats);
-      setPlayerStats(stats);
+      props.postTankStats(accountId);
     })();
   }, [accountId]);
 
-  const tankStats = playerStats.data ? playerStats.data.find((tank) => tank.tank_id === Number(wotId)) : [];
+  const tankStats = props.tanksStats.data ? props.tanksStats.data.find((tank) => tank.tank_id === Number(wotId)) : [];
```

Comment:
You can use optional chaining (?.) to simplify code and do not duplicate `props.tankStats.data` twice
```suggestion
  const tankStats = props.tanksStats?.data?.find((tank) => tank.tank_id === Number(wotId)) || [];
```